polygeist/tools/cgeist/driver.cc

```diff
@@ -1040,70 +1062,94 @@ int main(int argc, char **argv) {
     return ExecuteCC1Tool(Argv);
   }
 
+  // Forward '-mllvm' arguments to LLVM.
+  {
+    SmallVector<const char *> Args;
+    for (int i = 0; i < argc; i++)
+      Args.push_back(argv[i]);
+
+    ArrayRef<const char *> Argv = makeArrayRef(Args);
+    const OptTable &OptTbl = getDriverOptTable();
+    const unsigned IncludedFlagsBitmask = options::CC1AsOption;
+    unsigned MissingArgIndex, MissingArgCount;
+    InputArgList InputArgs = OptTbl.ParseArgs(
+        Argv, MissingArgIndex, MissingArgCount, IncludedFlagsBitmask);
+
+    SmallVector<const char *> NewArgv = {Argv[0]};
+    for (const opt::Arg *InputArg :
+         InputArgs.filtered(driver::options::OPT_mllvm))
+      NewArgv.push_back(InputArg->getValue());
+    llvm::cl::ParseCommandLineOptions(NewArgv.size(), &NewArgv[0]);
+  }
+
   // Split up the arguments into MLIR and linkage arguments.
   SmallVector<const char *> LinkageArgs, MLIRArgs;
   const CgeistOptions Options =
       splitCommandLineOptions(argc, argv, LinkageArgs, MLIRArgs);
   assert(!MLIRArgs.empty() && "MLIRArgs should not be empty");
 
   // Register any command line options.
-  registerPassManagerCLOptions();
-  registerDefaultTimingManagerCLOptions();
+  mlir::registerMLIRContextCLOptions();
+  mlir::registerPassManagerCLOptions();
+  mlir::registerAsmPrinterCLOptions();
+  mlir::registerDefaultTimingManagerCLOptions();
 
   // Parse command line options.
-  llvm::cl::list<std::string> inputFileNames(
+  llvm::cl::list<std::string> InputFileNames(
       llvm::cl::Positional, llvm::cl::OneOrMore,
       llvm::cl::desc("<Specify input file>"), llvm::cl::cat(ToolOptions));
 
-  llvm::cl::list<std::string> inputCommandArgs(
+  llvm::cl::list<std::string> InputCommandArgs(
       "args", llvm::cl::Positional, llvm::cl::desc("<command arguments>"),
       llvm::cl::ZeroOrMore, llvm::cl::PositionalEatsArgs);
 
-  int size = MLIRArgs.size();
-  const char **data = MLIRArgs.data();
-  InitLLVM y(size, data);
-  llvm::cl::ParseCommandLineOptions(size, data);
+  // Register command line options specific to cgeist.
+  {
+    int Size = MLIRArgs.size();
+    const char **Data = MLIRArgs.data();
+    InitLLVM Y(Size, Data);
```

Comment:
Why are you putting the `InitLLVM` object in a scope? 
We probably need it to be in the same scope as `main()` for `-debug-only=` to work.

Reply:
done